### DISCUSSION TITLE
How to subclass a table?

### DISCUSSION THREAD

**Discussion by ftCLI** [fonttools/fonttools#2796] [Q&A] [answered]
Is it possible to subclass a table to add methods?

For example:

`OS2.setBoldBit()` or `name.fromCsv()`

Actually I subclassed TTFont and use `self[OS/2].setBoldBit()` and so on.

**Comment by justvanrossum** [comment, marked as answer]
You can register a subclass with `fontTools.ttLib.ttFont.registerCustomTableClass()`; it will override the default class.

**Comment by ftCLI** [comment]
Thanks for the reply. I'm doing something wrong.

My goal is to add methods to `table_O_S_2f_2`

For example:

    registerCustomTableClass("OS/2", "ftCLI.Lib.OS_2")

    class TableOS2(table_O_S_2f_2):

        def customFunction(self):
            print(self.__dict__)

If I run the following:

    font = TTFont("font.ttf")
    font["OS/2"].customFunction()

I get an attribute error:

    AttributeError: 'table_O_S_2f_2' object has no attribute 'customFunction'

**Reply by justvanrossum** [reply]
Try
```python
registerCustomTableClass("OS/2", "ftCLI.Lib.OS_2", "TableOS2")
```

If you don't provide the class name as the third argument, it will fall back to `"table_O_S_2f_2"`, which you nicely imported in your module, so effectively it took the built-in class.

**Comment by ftCLI** [comment]
Thanks indeed! It works fine now.
